osu.Game/Screens/Play/BreaksOverlay/RemainingTimeCounter.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (c) 2007-2017 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using osu.Framework.Graphics.Containers;
+using osu.Game.Graphics.Sprites;
+using osu.Framework.Graphics;
+using System;
+
+namespace osu.Game.Screens.Play.BreaksOverlay
+{
+    public class RemainingTimeCounter : Container
+    {
+        private readonly OsuSpriteText counter;
+
+        private int? previousSecond;
+
+        private double remainingTime;
+
+        private bool isCounting;
+
+        public RemainingTimeCounter()
+        {
+            AutoSizeAxes = Axes.Both;
+            Alpha = 0;
+            Child = counter = new OsuSpriteText
+            {
+                Anchor = Anchor.Centre,
+                Origin = Anchor.Centre,
+                TextSize = 33,
+                Font = "Venera",
+            };
+        }
+
+        public void StartCounting(double remainingTime)
+        {
+            this.remainingTime = remainingTime;
+            isCounting = true;
+        }
+
+        protected override void Update()
+        {
+            base.Update();
+
+            if (isCounting)
+            {
+                var currentTime = Clock.CurrentTime;

```

Comment:
pretty sure this logic is wrong (the test case doesn't work as expected). `StartCounting` should probably accept and end time, not a remaining time.